docker.Makefile

```diff
@@ -110,7 +110,13 @@ lint: ## run linter(s)
 
 vendor: build_dev_image
 	$(info Update Vendoring...)
-	docker run --rm -v "$(CURDIR)":/go/src/github.com/docker/app/ $(DEV_IMAGE_NAME) make vendor
+	rm -rf ./vendor
+	docker rm -f docker-app-vendoring || echo ""
+	docker create --name docker-app-vendoring -v docker-app-vendor-cache://dep-cache -e DEPCACHEDIR=//dep-cache $(DEV_IMAGE_NAME) sh -c "rm -rf ./vendor && make vendor"
```

Comment:
why the double `//`?

Reply:
mingw flavor of sh (which is used under the hood by make) tries to be clever and when calling a non-mingw executable (not sure how it does the distinction between vanilla win32 executable and those compiled in a mingw environment though), tries to expand any argument that looks like a unix path into the equivalent windows path. docker CLI is not a mingw executable, so it expands things like /dep-cache into `c:\<path to mingw root\dep-cache`. Using the double `//` disable this expansion mechanism.

Reply:
:open_mouth:, thanks for explaining!

Reply:
Is the `rm -rf ./vendor` needed? I see the makefile already removes;

https://github.com/docker/app/blob/151e1c59c85940d692dd317058950838be1555b2/Makefile#L109

Reply:
@simonferquel would be good to add a comment about the double slashes.

Reply:
Just to check, there's no amount of `"quoting"` or `'quoting'` which will cause mingw to not do this, is there? (If there is then this would look much less unnatural...)

Reply:
I think setting `MSYS_NO_PATHCONV=1` disables the behavior

Reply:
see 
https://github.com/moby/moby/issues/12590#issuecomment-294141429 (and https://stackoverflow.com/a/34386471/1811501)

(and some other tickets https://github.com/moby/moby/search?q=MSYS_NO_PATHCONV&type=Issues)

Reply:
MSYS_NO_PATHCONV=1 disables this behavior with git bash. It does not seem to do it on my cygwin environment. 
The only thing that seems to uniformally work (without causing problems for mounting), is the double "//".

I will comment on that

Reply:
the rm -rf ./vendor has to happen locally, so that we reset it with the equivalent coming from the container results (if we don't do it, we won't properly delete files that have been removed by the new vendoring)

Reply:
It goes further than that:
```console
$ touch vendor/github.com/docker/cli/cli/deleteme.go
$ dep ensure -v
# Bringing vendor into sync
(1/1) Wrote github.com/docker/cli (from https://github.com/chris-crone/cli)@d6bfd7e5592dad85969516c131d33910fa5ebd58: hash of vendored tree didn't match digest in Gopkg.lock
```

so if the package exists in `vendor/` it _insists_ that it is exactly what it wrote there previously (rather than, say, fixing it like you probably wanted it to!). I don't see any flag which changes this behaviour.